src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/HotReload/ProjectHotReloadSession.cs

```diff
@@ -174,6 +175,14 @@ public async Task StartSessionAsync(CancellationToken cancellationToken)
         DebugTrace($"Start session for project '{_configuredProject.UnconfiguredProject.FullPath}' with TFM '{targetFramework}' and HotReloadRestart {runningProjectInfo.RestartAutomatically}");
 
         var processInfo = new ManagedEditAndContinueProcessInfo();
+
+        // If the debugger uses ICorDebug, the debugger is responsible for applying deltas to the process and the Hot Reload client receives empty deltas.
+        // Otherwise, the Hot Reload client is responsible for applying deltas and needs to receive them from the debugger.
+        // This is controlled by IsDebuggedProcess flag.
+        var flags = _debugLaunchOptions.HasFlag(DebugLaunchOptions.NoDebug) || await UsingLegacyWebAssemblyDebugEngineAsync(cancellationToken)
```

Comment:
For WASM + new ICorDebug, the delta will be applied using hotreload client even in debug mode?